addons/sale_timesheet/controllers/portal.py

```diff
@@ -63,6 +64,24 @@ def _get_groupby_mapping(self):
             invoice='timesheet_invoice_id')
         return groupby_mapping
 
+    def _task_get_page_view_values(self, task, access_token, **kwargs):
+        values = super()._task_get_page_view_values(task, access_token, **kwargs)
+        values['so_accessible'] = False
+        try:
+            if task.sale_order_id and self._document_check_access('sale.order', task.sale_order_id.id):
+                values['so_accessible'] = True
+        except (AccessError, MissingError):
+            pass
+
+        values['invoices_accessible'] = []
+        for invoice in task.sale_order_id.invoice_ids:
+            try:
+                if(self._document_check_access('account.move', invoice.id)):
```

Comment:
The parenthesis are not required like other programming language, you can write the condition like this if the parenthesis are not necessary. 🙂 
```suggestion
                if self._document_check_access('account.move', invoice.id):
```